src/api/utils/index.js

```diff
@@ -1,31 +1,117 @@
+import 'babel-polyfill';
 import imm from 'object-path-immutable';
+import Registry from '../../registry';
 
 export apiClient from './api-client';
 
+/**
+ * @module
+ * @private
+ */
+
+/**
+ * @example
+ * incrementProperty(state, 'isReading')
+ * @param {Object} state Object on which the key is incremented
+ * @param {string} key Key to increment
+ */
 export const incrementProperty = (state, key) => (
   imm.set(state, key, state[key] + 1)
 );
 
+/**
+ * @example
+ * decrementProperty(state, 'isReading')
+ * @param {Object} state Object on which the key is decremented
+ * @param {string} key Key to decrement
+ */
 export const decrementProperty = (state, key) => (
   imm.set(state, key, state[key] - 1)
 );
 
-export const resourcesArray = (data) => (
+/**
+ * @param {Array|Object} data
+ */
+export const resourcesArray = (data = []) => (
   Array.isArray(data) ? data : [data]
 );
 
+/**
+ * @example
+ * const obj = { foo: bar: { baz: 'hi' } }
+ * hasProperty(obj, 'foo', 'bar', 'baz') // => 'hi'
+ * hasProperty(obj, 'a', 'b', 'c') // => undefined
+ *
+ * @param {Object} object Object to traverse
+ * @param {String[]} keys Keys to look up
+ */
+export const hasProperty = (object, ...keys) => (
```

Comment:
Yeah, that makes sense. Should be pretty easy to migrate off `yup` and we can just use MDN's `Object#values` polyfill since that's the only non-transpilable feature we're using.

Can we change to `getProperty` then? I would think `hasProperty` should return a boolean :D